### DISCUSSION TITLE
Add flag to v fmt to set max line length

### DISCUSSION THREAD

**Discussion by btiffin** [vlang/v#14655] [Ideas]
It would be nice to restrict line lengths generated by `v fmt` to make it easier to include listings in technical documentation and when targeting printed book width limitations.  Perhaps `-m max` or `-s size`?

**Comment by effinsky** [comment]
absolutely! I now work in GO, and one of the things that makes us conflicted in the line length issue. We have no good formatting solution, and gofmt says "you deal with it". So the promise that GO makes through gofmt that there'd be no more "bikeshedding" is really fairly empty. Well, thanks. So we have all these back-and-forths. Meanwhile, in other popular languages, Rust and JS included, it's standard, be it through rustfmt or prettier. 

So it would be much appreciated. 

And I'm already plugging V in my company :D:D:D perhaps a microservice at some point!

**Reply by medvednikov** [reply]
Good luck with using V at your company!

This is a very reasonable request, but the problem is no matter how useful the flag is, it does introduce variability and inconsistency between code bases.

Although you could argue that using different line lengths manually is also inconsistent.

@spytheman what do you think?

**Reply by effinsky** [reply]
it's shaping up to be a lovely language. 

Well, if you don't set a line length for the fmt to wrap at, it'll be a free for all, and there'll be inconsistency for sure.  I'd say make the rules consistent and then either 
1) let the folks decide on a line length that's going to guide how the fmt applies those rules, or 
2) at least set the fmt to some reasonable compromise value like 100 chars.

For some context, with GO now we're seeing golines getting developed to fill the gap left by gofmt. So the same thing will happen for V, I'm sure, at some point. And thus there are limits to the uniformity of code style with a formatter like that. 

Thing is, if you consider accounting for max line length, at least YOU control how that happens, imo.

Keep up the good work <3

**Reply by spytheman** [reply]
V fmt already does wrap lines.

There will be *no options* for vfmt. That is the whole point - it is opinionated, *not* configurable, and distributed with V itself.

People/companies are free to implement their own formatters, as separate projects if they wish, but the default one, that will be used by 99% of users, will never have options, ensuring consistent and mostly readable V codebases everywhere.

**Reply by spytheman** [reply]
> you could argue that using different line lengths manually is also inconsistent.

That is true, but so far it is mostly working fine 🤷🏻‍♂️ .

I would have preferred for the wrapping to happen at 120 characters, but that is kind of the point - different people have different preferences, and whatever the limit is, it will be always a compromise... I much prefer to accept the shorter wrapping limit, than to participate in bike shedding discussions every time I switch a project, about what the flag or config file option should be, if it was configurable.

Besides, if a line is too long, it is better for maintainability of a project, to split it manually (by using temporary variables for example), long before encountering that wrapping limit.

**Reply by spytheman** [reply]
> So the promise that GO makes through gofmt that there'd be no more "bikeshedding" is really fairly empty

My experience is the exact opposite. I've never argued with people about formatting in Go projects, precisely because Go has gofmt, it is easy to use, and everyone uses it.

**Reply by medvednikov** [reply]
@effinsky indeed vfmt does already wrap lines at 100 chars. Many people use multiple columns of text on one screen, that's why it's relatively low. (Hey at least it's not 80 chars 🙂)

Of course the developer is free to wrap the lines at fewer chars.

**Reply by effinsky** [reply]
@medvednikov I'm glad to hear that line length wrapping is already included at 100. Did not know that. Personally I love short lines and would not mind 80, but I know it's harder to abide by and there are folks like @spytheman that like 120 and such. So 100 is a nice compromise.

@spytheman well, my experience is the opposite. I joined projects where people SKIPPED these discussions altogether, resulting in lines up to and over 200chars. This was wholly unreadable, but, as with so many other things, some got used to that. When these discussions ensued later, it was hard at that point to do much to clean up the formatting. 

As for the current implementation of line wrapping/unwrapping, I'll bring your attention to this here ticket I created: https://github.com/vlang/v/issues/15545
